### PR TITLE
 Add support for min_cpu_platform in google_compute_instance.

### LINKED ISSUE #106: Add min_cpu_platform to google_compute_instance resource
_This issue was originally opened by @nodesocket as hashicorp/terraform#15038. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Google Compute Engine just released ([blog post](https://cloudplatform.googleblog.com/2017/05/Compute-Engine-updates-bring-Skylake-GA-Extended-Memory-and-more-VM-flexibility.html), [technical doc](https://cloud.google.com/compute/docs/instances/specify-min-cpu-platform)) a new feature which allows you specify the cpu platform when creating VM instances. This allow you to provision the newest Skylake cpus.

Therefore I suggest adding a new string option to the `google_compute_instance` resource called `cpu_platform`.

### PR DESCRIPTION
**Pull request by rosbo**
Fixes #106

### CODE REVIEW

File: google/resource_compute_instance_test.go [hunk 1]
```diff
@@ -1099,6 +1120,16 @@ func testAccCheckComputeInstanceHasGuestAccelerator(instance *computeBeta.Instan
 	}
 }
 
+func testAccCheckComputeInstanceHasMinCpuPlatform(instance *computeBeta.Instance, minCpuPlatform string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if instance.MinCpuPlatform != minCpuPlatform {
+			return fmt.Errorf("Wrong minimum CPU platform: expected %d, got %d", minCpuPlatform, instance.MinCpuPlatform)
```
Comment (selmanj) [inline google/resource_compute_instance_test.go]: Go vet caught an error here about using `%d` instead of `%s`

**Comment by selmanj** [comment]
I meant expose `cpu_platform` as well rather than make `min_cpu_platform` computed

**Comment by hamann** [comment]
Just for interest, where did you find `cpu_platform`? Or is that just a output param?

**Comment by munnerz** [comment]
I'm trying to create a managed instance group on GCE - I can't see any way to specify the min_cpu_platform on there however, either through the `google_compute_instance_group_manager` resource or `google_compute_instance_template`.

Am I missing something, or does this need updating on the resource type? 😄

**Comment by rosbo** [comment]
This issue was about adding support for `min_cpu_platform` only for the `google_compute_instance`. Issue #516 is for adding it to `google_compute_instance_template`.

Once #639 is in. I will tackled #516

**Pull request merged**